src/master-playlist-controller.js

```diff
@@ -429,15 +429,16 @@ export class MasterPlaylistController extends videojs.EventTarget {
       // This is because in LIVE, the player plays 3 segments from the end of the
       // playlist, and if `BUFFER_LOW_WATER_LINE` is greater than the duration availble
       // in those segments, a viewer will never experience a rendition upswitch.
-      if (!currentPlaylist.endList ||
-          // For the same reason as LIVE, we ignore the low water line when the VOD
-          // duration is below the max potential low water line
-          this.duration() < Config.MAX_BUFFER_LOW_WATER_LINE ||
-          // we want to switch down to lower resolutions quickly to continue playback, but
-          nextPlaylist.attributes.BANDWIDTH < currentPlaylist.attributes.BANDWIDTH ||
-          // ensure we have some buffer before we switch up to prevent us running out of
-          // buffer while loading a higher rendition.
-          forwardBuffer >= bufferLowWaterLine) {
+      if (nextPlaylist &&
```

Comment:
perhaps this should just be a "return" if statement:

```js
if (!nextPlaylist || !nextPlaylist.attributes || !currentPlaylist || currentPlaylist.attributes) {
  return
}

Reply:
could be, though, I think we'd still want to trigger the `bandwidthupdate` event and it depends on what order we want to do that event and the media call.
If order doesn't matter, we can move the trigger above the check.

Reply:
that is true, I missed the `bandwidthupdate` triggering at the bottom, perhaps we should just have two event handlers and the second one should trigger the event, seems like we will always trigger it currently.